### PR TITLE
Fix travis script for running tests and fix broken tests

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Travis indicates a PR is valid event if legacy tests fail because the travis script is wrong
| Type?         | bug fix
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis tests should be green **AND** the travis logs should display no errors

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11898)

### CODE REVIEW

**Comment by matks** [comment]
## How to run Travis locally (using docker) to reproduce failing unit tests

```
$ docker run --name travis-debug -dit travisci/ci-garnet:packer-1512502276-986baf0 /sbin/init
$ docker exec -it travis-debug bash -l 
// in docker
$ su - travis
// as travis user
$ git clone --depth=50 https://github.com/PrestaShop/PrestaShop.git PrestaShop/PrestaShop
$ cd PrestaShop/PrestaShop/
// 11898 is PR number
$ git fetch origin +refs/pull/11898/merge
$ git checkout -qf FETCH_HEAD
$ cp travis-scripts/.composer-auth.json ~/.composer/auth.json
$ bash travis-scripts/setup-php-fpm.sh
$ bash travis-scripts/setup-apache.sh
$ cp tests-legacy/parameters.yml.travis app/config/parameters.yml
$ composer install --prefer-dist --no-interaction
$ bash travis-scripts/install-prestashop
$ bash tests-legacy/check_phpunit.sh
```

File: tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php [hunk 1]
```diff
@@ -51,16 +51,17 @@ public function setUp()
 
         /** @var ModuleManager */
         $moduleManager = self::$kernel->getContainer()->get('prestashop.module.manager');
-        if (!$moduleManager->isInstalled('bankwire')) {
-            $moduleManager->install('bankwire');
+
+        if (!$moduleManager->isInstalled('ps_banner')) {
```
Comment (PierreRambaud) [inline tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php:55]: trolololo

**Review by PierreRambaud — DISMISSED** [review]

File: classes/module/Module.php [hunk 2]
```diff
@@ -1982,28 +1982,45 @@ public function l($string, $specific = false, $locale = null)
...
                 }
             }
 
-            if (!isset($k) || !isset($res[$k]) || !isset($res[$k + 1])) {
+            if (!isset($thisModulePositionNumber)) {
+                // could not find hook positions for this module
+                return false;
+            }
+            if (!isset($sqlResult[$thisModulePositionNumber])) {
+                // ok this one is really weird
```
Comment (eternoendless) [inline classes/module/Module.php:2014]: 🤣

Comment (PierreRambaud) [reply classes/module/Module.php:2014]: Don't ask why.. We don't know :sob:

File: classes/module/Module.php [hunk 3]
```diff
@@ -1982,28 +1982,45 @@ public function l($string, $specific = false, $locale = null)
     public function updatePosition($id_hook, $way, $position = null)
     {
         foreach (Shop::getContextListShopID() as $shop_id) {
-            $sql = 'SELECT hm.`id_module`, hm.`position`, hm.`id_hook`
+            $sqlQueryToGetAvailableHookPositions = 'SELECT hm.`id_module`, hm.`position`, hm.`id_hook`
```
Comment (eternoendless) [inline classes/module/Module.php]: Your SQL variables could be less long if you inverted the `sql` part:

`$sqlQueryToGetXYZ` ➡️ `$getXYZSql`

File: classes/module/Module.php [hunk 4]
```diff
@@ -2012,20 +2029,20 @@ public function updatePosition($id_hook, $way, $position = null)
             $minPosition = min((int) $from['position'], (int) $to['position']);
             $maxPosition = max((int) $from['position'], (int) $to['position']);
 
-            $sql = 'UPDATE `' . _DB_PREFIX_ . 'hook_module`
+            $sqlQueryToShiftHookPositions = 'UPDATE `' . _DB_PREFIX_ . 'hook_module`
```
Comment (eternoendless) [inline classes/module/Module.php]: This could be renamed to `$shiftHookPositionsSql`

File: classes/module/Module.php [hunk 5]
```diff
@@ -2012,20 +2029,20 @@ public function updatePosition($id_hook, $way, $position = null)
...
                 WHERE position BETWEEN ' . $minPosition . ' AND ' . $maxPosition . '
                 AND `id_hook` = ' . (int) $from['id_hook'] . ' AND `id_shop` = ' . $shop_id;
 
-            if (!Db::getInstance()->execute($sql)) {
+            if (!Db::getInstance()->execute($sqlQueryToShiftHookPositions)) {
                 return false;
             }
 
-            $sql = 'UPDATE `' . _DB_PREFIX_ . 'hook_module`
+            $sqlQueryToCreateMissingPosition = 'UPDATE `' . _DB_PREFIX_ . 'hook_module`
```
Comment (eternoendless) [inline classes/module/Module.php]: And this one to `$createMissingPosition`

File: src/Adapter/Module/ModuleDataProvider.php [hunk 6]
```diff
@@ -182,7 +182,21 @@ public function isEnabled($name)
     public function isInstalled($name)
     {
         // ToDo: Load list of all installed modules ?
-        return (bool) Db::getInstance()->getValue('SELECT `id_module` FROM `' . _DB_PREFIX_ . 'module` WHERE `name` = "' . pSQL($name) . '"');
+        return (bool) $this->getModuleIdByName($name);
+    }
+
+    /**
+     * Return the Module Id
```
Comment (eternoendless) [inline src/Adapter/Module/ModuleDataProvider.php]:
```suggestion
     * Returns the Module Id
```

File: src/Adapter/Module/ModuleDataProvider.php [hunk 7]
```diff
@@ -182,7 +182,21 @@ public function isEnabled($name)
...
+     *
+     * @param string $name The technical module name
+     *
+     * @return int The Module Id
```
Comment (eternoendless) [inline src/Adapter/Module/ModuleDataProvider.php]:
```suggestion
     * @return int The Module Id, or 0 if not found.
```

File: src/Core/Addon/Module/ModuleManager.php [hunk 8]
```diff
@@ -699,6 +699,19 @@ public function isInstalled($name)
         return $this->moduleProvider->isInstalled($name);
     }
 
+    /**
+     * Shortcut to the module data provider in order to know the module id depends
+     * on its name.
+     *
+     * @param string $name The technical module name
+     *
+     * @return int The module Id
```
Comment (eternoendless) [inline src/Core/Addon/Module/ModuleManager.php]:
```suggestion
     * @return int The module Id, or 0 if not found.
```

File: tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php (hunk 1 above)
Comment (eternoendless) [reply tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php:55]: Why change the module in question?

File: tests-legacy/Integration/classes/module/ModuleOverrideInstallUninstallTest.php [hunk 9]
```diff
@@ -56,8 +56,8 @@ protected function setUp()
         $this->moduleManager = $this->moduleManagerBuilder->build();
 
         $this->moduleNames= [
-            'pscsx3241',
             'pscsx32412',
+            'pscsx3241',
```
Comment (eternoendless) [inline tests-legacy/Integration/classes/module/ModuleOverrideInstallUninstallTest.php:60]: Why was the order inverted?

File: tests-legacy/Unit/ContextMocker.php [hunk 10]
```diff
@@ -101,7 +101,7 @@ public function mockContext()
         Phake::when($context->customer)->getGroups()->thenReturn(array());
         $context->cookie   = Phake::mock('Cookie');
         $context->country  = Phake::mock('Country');
-        $context->language = new Language((int) Configuration::get('PS_LANG_DEFAULT'));
+        $context->language = new Language((int) Configuration::get('PS_DETECT_LANG'));
```
Comment (eternoendless) [inline tests-legacy/Unit/ContextMocker.php]: Why this change?

File: tests-legacy/resources/ModulesOverrideInstallUninstallTest/AdminProductsController.php [hunk 11]
```diff
@@ -1,280 +1,11 @@
...
-                'align' => 'center',
-                'position' => 'position',
-            );
-        }
-    }
-
     /*
     * module: pscsx32412
-    * date: 2015-07-13 16:05:31
+    * date: 2018-12-26 14:14:05
```
Comment (eternoendless) [inline tests-legacy/resources/ModulesOverrideInstallUninstallTest/AdminProductsController.php:8]: What's this?

File: classes/module/Module.php (hunk 2 above)
Comment (matks) [reply classes/module/Module.php:2014]: I kept it because it was the legacy behavior but this case is logically impossible.

File: tests-legacy/resources/ModulesOverrideInstallUninstallTest/AdminProductsController.php (hunk 11 above)
Comment (PierreRambaud) [reply tests-legacy/resources/ModulesOverrideInstallUninstallTest/AdminProductsController.php:8]: This is something which is added when you add override in your module.

File: tests-legacy/Unit/ContextMocker.php (hunk 10 above)
Comment (PierreRambaud) [reply tests-legacy/Unit/ContextMocker.php]: There is no value in database for `PS_LANG_DEFAULT`

File: tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php (hunk 1 above)
Comment (matks) [reply tests-legacy/Integration/PrestaShopBundle/Controller/Api/Improve/Design/PositionsControllerTest.php:55]: I struggled several hours to fix PositionsControllerTest::testMoveHookPositionToTop . Looks like `ps_featuredproducts` is in a strange state: it is installed according to ModuleManager but its hooks are not registered. Tried uninstall and install again, did not work.

I replaced the use of `ps_featuredproducts` (the git diff makes it hard to see but if you compare old/new version it's obvious) for this test by `ps_banner` and it solves the issue.

I think something is wrong with this module, maybe another test modified it / installed it and the fixture state is corrupted.

File: tests-legacy/Integration/classes/module/ModuleOverrideInstallUninstallTest.php (hunk 9 above)
Comment (PierreRambaud) [reply tests-legacy/Integration/classes/module/ModuleOverrideInstallUninstallTest.php:60]: Something was broken somewhere in legacy tests :/ Inverted make it work.. It's magic :/

File: tests-legacy/Unit/ContextMocker.php (hunk 10 above)
Comment (eternoendless) [reply tests-legacy/Unit/ContextMocker.php]: How come? That's usually the first setting to be inserted in DB

Comment (PierreRambaud) [reply tests-legacy/Unit/ContextMocker.php]: I knew it was weird. I try to restore the old value. Waiting for ci.

File: src/Adapter/Module/ModuleDataProvider.php (hunk 7 above)
Comment (PierreRambaud) [reply src/Adapter/Module/ModuleDataProvider.php]: Done.

File: src/Core/Addon/Module/ModuleManager.php (hunk 8 above)
Comment (PierreRambaud) [reply src/Core/Addon/Module/ModuleManager.php]: Done.

File: src/Adapter/Module/ModuleDataProvider.php (hunk 6 above)
Comment (PierreRambaud) [reply src/Adapter/Module/ModuleDataProvider.php]: Done.

**Comment by matks** [comment]
@eternoendless Requested changes have been applied ✅

**Review by eternoendless — APPROVED** [review]
LGTM

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @matks
